### PR TITLE
allow date and time to be adjusted externally after mount

### PR DESCRIPTION
**Pull request by darcyadams**

### CODE REVIEW

**Comment by darcyadams** [comment]
because the date picker relies on it's parent setting state onchange, we can't rely exclusively on passed props like we do for other inputs. This just implements a `componentWillreceiveProps` method that check if the parent to our component has modified the date passed back from `getValue` and updates the internal state to match.
This is a react anti-pattern, but I don't see a way around it while we are using this 3rd party date picker.

File: dist/components/date_picker.js [hunk 1]
```diff
@@ -27,9 +27,21 @@
       tabIndex: React.PropTypes.number,
       selected: React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.object])
     },
+    getDefaultProps: function() {
+      return {
+        placeholderText: 'Select a date',
```
Comment (ekaspern) [inline dist/components/date_picker.js]: do you want to translate this?

Comment (darcyadams) [inline dist/components/date_picker.js]: no, user can pass a translated value, this is just the default

**Comment by ekaspern** [comment]
+1

**Pull request merged**